src/SassBuilder.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the SymfonyCasts SassBundle package.
+ * Copyright (c) SymfonyCasts <https://symfonycasts.com/>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfonycasts\SassBundle;
+
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Process\InputStream;
+use Symfony\Component\Process\Process;
+
+class SassBuilder
+{
+    private ?SymfonyStyle $output = null;
+
+    /**
+     * @param array<string> $sassPaths
+     */
+    public function __construct(
+        private readonly array $sassPaths,
+        private readonly string $cssPath,
+        private readonly string $projectRootDir,
+        private readonly ?string $binaryPath
+    ) {
+    }
+
+    public function runBuild(bool $watch): Process
+    {
+        $binary = $this->createBinary();
+
+        $args = $this->getScssCssTargets();
+
+        if ($watch) {
+            $args[] = '--watch';
+        }
+
+        $process = $binary->createProcess($args);
+
+        if ($watch) {
+            $process->setTimeout(null);
+
+            $inputStream = new InputStream();
+            $process->setInput($inputStream);
+        }
+
+        $this->output?->note('Executing Sass (pass -v to see more details).');
+        if ($this->output?->isVerbose()) {
+            $this->output->writeln([
+                '  Command:',
+                '    '.$process->getCommandLine(),
+            ]);
+        }
+
+        $process->start();
+
+        return $process;
+    }
+
+    /**
+     * @return array<string>
+     */
+    public function getScssCssTargets(): array
+    {
+        $targets = [];
+        foreach ($this->sassPaths as $sassPath) {
+            $targets[] = $sassPath . ':' . ScssFileNameGuesser::guessCssNameFromSassFile($sassPath, $this->cssPath);
```

Comment:
Minor: let's inline the `guessCssNameFromSassFile()` into this class, make it `public` (non-static, see below) with`@internal` on it. Call the method: `getSassOutputFilename()`. 

Then, we inject `SassBuilder` into the compiler class. This will avoid needing to pass the scss paths around to multiple classes: make `SassBuilder` the central class that's aware of this, and give it any public methods it needs to fetch off that info. It just seems a bit simpler - we do it in TailwindBundle - https://github.com/SymfonyCasts/tailwind-bundle/blob/main/src/AssetMapper/TailwindCssAssetCompiler.php#L28